src/daemon/rpc.rs

```diff
@@ -104,107 +112,108 @@ impl rpc::VarlinkInterface for Server {
                 failure: None,
             },
-            Event::Completed { nix_file, result } => rpc::Event {
+            Event::Completed { .. } => proto::Event {
                 kind: kind::completed,
-                nix_file: Some(nix_file.try_into()?),
+                section: None,
                 reason: None,
-                result: Some(result.try_into()?),
+                result: Some(ev.try_into()?),
                 failure: None,
             },
-            Event::Failure { nix_file, failure } => rpc::Event {
+            Event::Failure { .. } => proto::Event {
                 kind: kind::failure,
-                nix_file: Some(nix_file.try_into()?),
+                section: None,
                 reason: None,
                 result: None,
-                failure: Some(failure.try_into()?),
+                failure: Some(ev.try_into()?),
             },
         })
     }
 }
 
-impl TryFrom<Event> for rpc::Event {
-    type Error = &'static str;
+impl TryFrom<build_loop::Event> for proto::Event {
+    type Error = String;
 
-    fn try_from(ev: Event) -> Result<Self, Self::Error> {
-        rpc::Event::try_from(&ev)
+    fn try_from(ev: build_loop::Event) -> Result<Self, Self::Error> {
+        proto::Event::try_from(&ev)
     }
 }
 
-impl TryFrom<rpc::Monitor_Reply> for Event {
+impl TryFrom<proto::Monitor_Reply> for build_loop::Event {
     type Error = String;
 
-    fn try_from(mr: rpc::Monitor_Reply) -> Result<Self, Self::Error> {
-        Event::try_from(mr.event)
+    fn try_from(mr: proto::Monitor_Reply) -> Result<Self, Self::Error> {
+        build_loop::Event::try_from(mr.event)
     }
 }
 
-impl TryFrom<rpc::Event> for Event {
+impl TryFrom<proto::Event> for build_loop::Event {
     type Error = String;
 
-    fn try_from(re: rpc::Event) -> Result<Self, Self::Error> {
-        use rpc::Event_kind::*;
+    fn try_from(re: proto::Event) -> Result<Self, Self::Error> {
+        use proto::Event_kind::*;
 
         Ok(match re.kind {
-            section_end => Event::SectionEnd,
-            started => Event::Started {
-                nix_file: re.nix_file.ok_or("missing nix file!")?.try_into()?,
-                reason: re.reason.ok_or("missing reason!")?.try_into()?,
-            },
-            completed => Event::Completed {
-                nix_file: re.nix_file.ok_or("missing nix file!")?.try_into()?,
-                result: re.result.ok_or("missing result!")?.into(),
-            },
-            failure => Event::Failure {
-                nix_file: re.nix_file.ok_or("missing nix file!")?.try_into()?,
-                failure: re.failure.ok_or("missing failure log")?.try_into()?,
-            },
+            section_end => build_loop::Event::SectionEnd,
+            started => re.reason.ok_or("missing reason!")?.try_into()?,
+            completed => re.result.ok_or("missing result!")?.try_into()?,
+            failure => re.failure.ok_or("missing failure log")?.try_into()?,
```

Comment:
missing exclamation mark? :)

Reply:
Actually, I don't think the exclamation marks are necessary. It's already in an error message. No need to draw more attention to it IMO.